src/base/cub_ddl_log.c

```diff
@@ -0,0 +1,1206 @@
+    {
+      cub_ddl_log_init ();
+      return;
+    }
+
+  FREE_MEM (ddl_audit_handle->sql_text);
+  FREE_MEM (ddl_audit_handle->err_msg);
+
+  ddl_audit_handle->stmt_type = -1;
+  ddl_audit_handle->execute_type = DDL_LOG_RUN_EXECUTE_FUNC;
+  ddl_audit_handle->loaddb_file_type = LOADDB_FILE_TYPE_NONE;
+  ddl_audit_handle->log_filepath[0] = '\0';
+  ddl_audit_handle->execute_start_time[0] = '\0';
+  ddl_audit_handle->elapsed_time[0] = '\0';
+  ddl_audit_handle->file_line_number = 0;
+  ddl_audit_handle->err_code = 0;
+  ddl_audit_handle->msg[0] = '\0';
+  ddl_audit_handle->file_name[0] = '\0';
+  ddl_audit_handle->schema_file[0] = '\0';
+  ddl_audit_handle->commit_count = 0;
+
+  if (all_free)
+    {
+      ddl_audit_handle->auto_commit_mode = FALSE;
+      is_executed_ddl = FALSE;
+    }
+}
+
+void
+cub_ddl_log_destroy ()
+{
+  if (ddl_audit_handle != NULL)
+    {
+      FREE_MEM (ddl_audit_handle->sql_text);
+      FREE_MEM (ddl_audit_handle->err_msg);
+      FREE_MEM (ddl_audit_handle);
+    }
+}
+
+void
+cub_ddl_log_app_name (T_APP_NAME app_name)
+{
+  if (ddl_audit_handle == NULL)
+    {
+      return;
+    }
+  ddl_audit_handle->app_name = app_name;
+}
+
+void
+cub_ddl_log_db_name (const char *db_name)
+{
+  char *pstr = NULL;
+  if (ddl_audit_handle == NULL || db_name == NULL)
+    {
+      return;
+    }
+
+  snprintf (ddl_audit_handle->db_name, sizeof (ddl_audit_handle->db_name), db_name);
+
+  pstr = (char *) strchr (ddl_audit_handle->db_name, '@');
+  if (pstr != NULL)
+    {
+      *pstr = '\0';
+    }
+}
+
+void
+cub_ddl_log_user_name (const char *user_name)
+{
+  if (ddl_audit_handle == NULL || user_name == NULL)
+    {
+      return;
+    }
+
+  snprintf (ddl_audit_handle->user_name, sizeof (ddl_audit_handle->user_name), user_name);
+}
+
+void
+cub_ddl_log_ip (const char *ip_addr)
+{
+  if (ddl_audit_handle == NULL || ip_addr == NULL)
+    {
+      return;
+    }
+
+  snprintf (ddl_audit_handle->ip_addr, sizeof (ddl_audit_handle->ip_addr), ip_addr);
+}
+
+void
+cub_ddl_log_pid (const int pid)
+{
+  if (ddl_audit_handle == NULL)
+    {
+      return;
+    }
+
+  ddl_audit_handle->pid = pid;
+}
+
+void
+cub_ddl_log_br_name (const char *br_name)
+{
+  if (ddl_audit_handle == NULL || br_name == NULL)
+    {
+      return;
+    }
+  snprintf (ddl_audit_handle->br_name, BROKER_NAME_LEN, br_name);
+}
+
+void
+cub_ddl_log_br_index (const int index)
+{
+  if (ddl_audit_handle == NULL)
+    {
+      return;
+    }
+  ddl_audit_handle->br_index = index;
+}
+
+void
+cub_ddl_log_sql_text (char *sql_text, int len)
+{
+  if (ddl_audit_handle == NULL || sql_text == NULL || len < 0)
+    {
+      return;
+    }
+
+  if (ddl_audit_handle->sql_text != NULL)
+    {
+      FREE_MEM (ddl_audit_handle->sql_text);
+    }
+
+  ddl_audit_handle->sql_text = (char *) MALLOC (len + 1);
+
+  memset (ddl_audit_handle->sql_text, 0x0, len + 1);
+
+  if (ddl_audit_handle->sql_text != NULL)
+    {
+      snprintf (ddl_audit_handle->sql_text, len + 1, sql_text);
```

Comment:
Isn't it
`snprintf (ddl_audit_handle->sql_text, len, sql_text);`
Copying 0x00 twice may not be needed.

Reply:
Thank you, i will change code as your recommend.

Reply:
Thank you, i will change code as your recommend.